microbench/src/main/java/io/netty/microbench/headers/HeadersBenchmark.java

```diff
@@ -139,4 +139,22 @@ public void http2Iterate(Blackhole bh) {
             bh.consume(entry);
         }
     }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void httpAddAllFast(Blackhole bh) {
+        bh.consume(new DefaultHttpHeaders(false).add(httpHeaders));
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void http2AddAllFast(Blackhole bh) {
+        bh.consume(new DefaultHttp2Headers(false).add(http2Headers));
+    }
+
+    @Benchmark
+    @BenchmarkMode(Mode.AverageTime)
+    public void http2AddAllSlow(Blackhole bh) {
+        bh.consume(new DefaultHttp2Headers().add(http2Headers));
```

Comment:
@louiscryan - Can you re-run with validation off (to be fair to other benchmarks)? `new DefaultHttp2Headers(false)`